lib/ansible/modules/cloud/amazon/cloudformation.py

```diff
@@ -361,6 +363,40 @@ def stack_operation(cfn, stack_name, operation):
             time.sleep(5)
     return {'failed': True, 'output':'Failed for unknown reasons.'}
 
+
+def get_changeset(module, stack_params, cfn):
+    """Create a change set, get it and delete it."""
+    randomchars = [random.choice(string.ascii_letters + string.digits) for x in range(0, 10)]
+    name = 'ansible-cloudformation-check-' + ''.join(randomchars)
+    try:
+        change_set = cfn.create_change_set(ChangeSetName=name, **stack_params)
```

Comment:
When running check mode and a stack does not yet exist, this call fails. If you catch botocore.exceptions.ClientError in a separate `except` you can check if the error is that the stack doesn't exist, then return None and show the state as "changed"